drv/opentitan-aontimer/src/lib.rs

```diff
@@ -0,0 +1,150 @@
+            (*inst.base_addr)[WDOG_BARK_THOLD] = inst.ms_to_reg(bark_thold_ms);
+            if bark_thold_ms > bite_thold_ms {
+                (*inst.base_addr)[WDOG_BITE_THOLD] =
+                    inst.ms_to_reg(bark_thold_ms);
+            } else {
+                (*inst.base_addr)[WDOG_BITE_THOLD] =
+                    inst.ms_to_reg(bite_thold_ms);
+            }
+        }
+        inst
+    }
+
+    //Disable the watchdog timer. If the configuration is locked, returns false.
+    pub fn disable(&self) -> bool {
+        if !self.check_lock() {
+            return false;
+        }
+        unsafe {
+            (*self.base_addr)[WDOG_CTRL] = 0;
+        }
+        true
+    }
+
+    //Enables the watchdog timer. If the configuration is locked, returns false.
+    pub fn enable(&self) -> bool {
+        if !self.check_lock() {
+            return false;
+        }
+        unsafe {
+            (*self.base_addr)[WDOG_CTRL] |= 1;
+        }
+        true
+    }
+
+    //Enables the watchdog timer, then locks the configuration. If the
+    // configuration is already locked, returns false.
+    pub fn enable_and_lock(&self) -> bool {
+        if !self.check_lock() {
+            return false;
+        }
+        unsafe {
+            (*self.base_addr)[WDOG_CTRL] |= 1;
+            (*self.base_addr)[WDOG_REGWEN] = 0;
+        }
+        true
+    }
+
+    //Sets the bark threshold in ms. If the configuration is locked, returns false.
+    pub fn set_bark_thold(&self, bark_thold_ms: u64) -> bool {
+        if !self.check_lock() {
+            return false;
+        }
+        unsafe {
+            (*self.base_addr)[WDOG_BARK_THOLD] = self.ms_to_reg(bark_thold_ms);
+        }
+        true
+    }
+
+    //Sets the bite threshold in ms. If the configuration is locked, returns false.
+    pub fn set_bite_thold(&self, bite_thold_ms: u64) -> bool {
+        if !self.check_lock() {
+            return false;
+        }
+        unsafe {
+            (*self.base_addr)[WDOG_BITE_THOLD] = self.ms_to_reg(bite_thold_ms);
+        }
+        true
+    }
+
+    //Pets the watchdog, i.e. resets the count down to 0.
+    pub fn feed_sacrifice(&self) {
+        unsafe {
+            (*self.base_addr)[WDOG_COUNT] = 0;
+        }
+    }
+
+    pub fn clear_wdt_irq(&self) {
+        unsafe {
+            (*self.base_addr)[INTR_STATE] |= 0x2;
+        }
+    }
+
+    //Checks whether the configuration is locked.
+    #[inline(always)]
+    fn check_lock(&self) -> bool {
+        unsafe { (*self.base_addr)[WDOG_REGWEN] == 1 }
+    }
+
+    // Converts  milliseconds to an appropriate count in the register.
+    // If the result is greater than max int, the value is saturated to max int.
+    fn ms_to_reg(&self, ms: u64) -> u32 {
+        let mut time: u64 = ms * ((self.clock_freq / 1000) as u64);
+        if time > u32::MAX.into() {
+            time = u32::MAX.into();
+        }
+        time.try_into().unwrap()
```

Comment:
Is silently truncating to `u32` the right behavior here?  Should we return a result so the caller knows the desired `ms` was too big?

Reply:
I think returning error would be better since silently assuming a value would mean that the caller doesn't realize that the timer expires sooner than intended. I am hoping we don't ever see that long of a timeout but might make sense to return appropriate error.

Reply:
Changed.